src/Equinor.ProCoSys.Completion.Command/Validators/DocumentValidator.cs

```
@@ -13,19 +13,18 @@ public class DocumentValidator : IDocumentValidator
 {
     private readonly IReadOnlyContext _context;
 
+    // Trick to write LINQ queries to let EF create effective SQL queries is
+    // 1) use Any
+    // 2) select a projection with as few columns as needed
     public DocumentValidator(IReadOnlyContext context) => _context = context;
 
     public async Task<bool> ExistsAsync(Guid documentGuid, CancellationToken cancellationToken) =>
-        await (from l in _context.QuerySet<Document>()
-            where l.Guid == documentGuid
-            select l).AnyAsync(cancellationToken);
-
-    public async Task<bool> IsVoidedAsync(Guid documentGuid, CancellationToken cancellationToken)
-    {
-        var document = await(from d in _context.QuerySet<Document>()
+        await (from d in _context.QuerySet<Document>()
             where d.Guid == documentGuid
-            select d).SingleOrDefaultAsync(cancellationToken);
+            select 1).AnyAsync(cancellationToken);
```

We lose the "fail fast" we get from SingleOrDefault when there are multiple results when we only expect 1 result.
Has this tradeoff been evaluated and concluded that the optimization is more worth than the "fail fast".

(I have been burnt more times than I care to count from a faulty query filter leading to multiple results and picking an incorrect result value)

---

This is more about how EF create the SQL. 
Using AnyAsync, EF produces DB queries as 
```
SELECT
    CASE WHEN EXISTS (
        SELECT 1
        FROM YourTableName AS t
        WHERE t.SomeColumn = @someValue
    ) THEN CAST(1 AS bit)
    ELSE CAST(0 AS bit) END
```
SingleOrDefaulltAsync produces DB query with top(2) of columns selected

EXISTS is recommended to check if item exists in DB.

I've measured 100 selects in a row for each method. Any seem to be slightly faster in my box